src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -605,97 +612,135 @@ public HttpResponse onSuccess(String authorizationCode) {
-        // the panel "Configure Global Security", new instances of Global/ProjectMatrixAuthorizationStrategy are created, where
-        // existing users are set up again.
-        // we'll mimic what the "Configure Global Security" config page does                                        
+                
+        // we append the role suffix to the name stored into Jenkins, since a given user is able to log in at varying scope/permission
+        // levels in openshift; however, for now, we make sure the display name for Jenkins does not include this suffix
+        String suffix = "";
+        if (allowedRoles.contains("admin"))
+            suffix = "-admin";
+        else if (allowedRoles.contains("edit"))
+            suffix = "-edit";
+        else if (allowedRoles.contains("view"))
+            suffix = "view";
+        String matrixKey = info.getName() + suffix;
         
-        User u = User.get(token.getName());
-        info.updateProfile(u);
+        // logs this user in.... with the index of UsernamePasswordAuthenticationToken token being matrixKey, that will tell jenkins auth
+        // code down the line what permissions to map to, where there 3 permissions for each user possible, where the key for that permission
+        // is the username appended by -admin, -edit, or -view
+        // NOTE, if all three self-sars fail, then we don't want to set any auth token in the jenkins security context;  that way, a malicious 
+        // user can't say create a "foo-admin" user, and get user foo's admin permission; we leave the jenkins anonymous user in the context,
+        // where that user has no permissions;  note, if "foo-admin" has say view access, then his permission key via this token and matrixKey
+        // will be "foo-admin-view", and only have the jenkins permissions we've assigned to the view role
+        UsernamePasswordAuthenticationToken token = null;
         
-        //NOTE, Jenkins currently does not employ any sort of synchronization around their paths for updating the authorization strategy;
-        //However, with user login now driving the addition of users and ther permissions, that does not seem prudent when users are 
-        //logging in concurrently.
-
-        synchronized(USER_UPDATE_LOCK) {
-        	GlobalMatrixAuthorizationStrategy existingAuthMgr = (GlobalMatrixAuthorizationStrategy) Jenkins.getInstance().getAuthorizationStrategy();
-        	Set<String> usersGroups = existingAuthMgr.getGroups();
-            List<PermissionGroup> permissionGroups = new ArrayList<PermissionGroup>(PermissionGroup.getAll());
-            GlobalMatrixAuthorizationStrategy newAuthMgr = null;
-            if (existingAuthMgr instanceof ProjectMatrixAuthorizationStrategy) {
-            	newAuthMgr = new ProjectMatrixAuthorizationStrategy();
-            } else {
-            	newAuthMgr = new GlobalMatrixAuthorizationStrategy();
-            }
+        if (suffix.length() != 0) {
```

Comment:
Yes a -none suffix would achieve similar affect.  I'll update to do it that way.